src/nvim/diff.c

```diff
@@ -1830,6 +2053,45 @@ int diff_check(win_T *wp, linenr_T lnum)
     return 0;
   }
 
+  if (dp->df_redraw && diff_linematch(dp)) {
+    run_linematch_algorithm(dp);
+  }
+
+  if (dp->is_linematched) {
+    int filler_lines_d1 = 0;
```

Comment:
Can this whole block be factored into a function like `int linematched_filler_lines(diff_T *dp, int idx, linenr_T lnum, int *linestatus)`?

Reply:
done